File: c/darts/src/darts.c
#include "darts.h"
#include <math.h>

uint8_t score(coordinate_t landing_position) {
    float distance = sqrt(pow(landing_position.x, 2) + pow(landing_position.y, 2));

    if (distance <= 1.0F) {
        return 10;
    } else if (distance <= 5.0F) {
        return 5;
    } else if (distance <= 10.0F) {
        return 1;
    } else {
        return 0;
    }
}

File: go/darts/darts.go
package darts

import "math"

func Score(x, y float64) int {
	distance := math.Sqrt(math.Pow(x, 2) + math.Pow(y, 2))

	switch {
	case distance <= 1.0:
		return 10
	case distance <= 5.0:
		return 5
	case distance <= 10.0:
		return 1
	default:
		return 0
	}
}
